contracts/VotingEscrow.vy

```diff
@@ -389,258 +256,162 @@ def checkpoint():
+    assert unlock_week > block.timestamp  #  dev: unlock time must be in the future
+    assert unlock_week <= block.timestamp + MAXTIME  # dev: voting lock can be 4 years max
+
+    self._deposit_for(msg.sender, msg.sender, amount, unlock_week, locked, CREATE_LOCK_TYPE)
 
-    self._deposit_for(msg.sender, _addr, _value, unlock_time, _locked, CREATE_LOCK_TYPE)
 
 @external
 @nonreentrant('lock')
-def increase_amount(_value: uint256):
+def increase_amount(amount: uint256):
     """
-    @notice Deposit `_value` additional tokens for `msg.sender`
+    @notice Deposit `amount` additional tokens for `msg.sender`
             without modifying the unlock time
-    @param _value Amount of tokens to deposit and add to the lock
+    @param amount Amount of tokens to deposit and add to the lock
     """
-    _locked: LockedBalance = self.locked[msg.sender]
+    locked: LockedBalance = self.locked[msg.sender]
 
-    assert _value > 0  # dev: need non-zero value
-    assert _locked.amount > 0, "No existing lock found"
-    assert _locked.end > block.timestamp, "Cannot add to expired lock. Withdraw"
+    assert amount > 0  # dev: need non-zero value
+    assert locked.amount > 0  # dev: no existing lock found
+    assert locked.end > block.timestamp  # dev: lock expired, call withdraw
 
-    self._deposit_for(msg.sender, msg.sender, _value, 0, _locked, INCREASE_LOCK_AMOUNT)
+    self._deposit_for(msg.sender, msg.sender, amount, 0, locked, INCREASE_LOCK_AMOUNT)
 
 
 @external
 @nonreentrant('lock')
-def increase_unlock_time(_unlock_time: uint256):
+def increase_unlock_time(unlock_time: uint256):
     """
-    @notice Extend the unlock time for `msg.sender` to `_unlock_time`
-    @param _unlock_time New epoch time for unlocking
+    @notice Extend the unlock time for `msg.sender` to `unlock_time`
+    @param unlock_time New epoch time for unlocking
     """
-    _locked: LockedBalance = self.locked[msg.sender]
-    unlock_time: uint256 = (_unlock_time / WEEK) * WEEK  # Locktime is rounded down to weeks
+    locked: LockedBalance = self.locked[msg.sender]
+    unlock_week: uint256 = unlock_time / WEEK * WEEK  # Locktime is rounded down to weeks
 
-    assert _locked.end > block.timestamp, "Lock expired"
-    assert _locked.amount > 0, "Nothing is locked"
-    assert unlock_time > _locked.end, "Can only increase lock duration"
-    assert unlock_time <= block.timestamp + MAXTIME, "Voting lock can be 4 years max"
+    assert locked.end > block.timestamp  # dev: lock expired
+    assert locked.amount > 0  # dev: nothing is locked
+    assert unlock_week > locked.end  # dev: can only increase lock duration
+    assert unlock_week <= block.timestamp + MAXTIME  # dev: voting lock can be 4 years max
 
-    self._deposit_for(msg.sender, msg.sender, 0, unlock_time, _locked, INCREASE_UNLOCK_TIME)
+    self._deposit_for(msg.sender, msg.sender, 0, unlock_week, locked, INCREASE_UNLOCK_TIME)
 
 
 @external
 @nonreentrant('lock')
 def withdraw():
```

Comment:
I don't think the withdraw function should be combined, some users might try to withdraw by mistake and lose a large amount of locked tokens.

Reply:
most users will use the contract from ui, those who don't usually read what it does

Reply:
maybe a previewWithdraw view function is useful here to calculate actual asset returned by executing a withdraw in the current state/block like in the erc4626 standard?

https://eips.ethereum.org/EIPS/eip-4626

Reply:
`withdraw` is a mutating function, it requires to take a snapshot to get the accurate value. maybe we can just make it return the withdrawn amount and penalty, then people would be able to call `veyfi.withdraw.call()`.